airflow/providers/amazon/aws/hooks/ecr.py

```diff
@@ -0,0 +1,101 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import base64
+import logging
+from dataclasses import dataclass
+from datetime import datetime
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.utils.log.secrets_masker import mask_secret
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class EcrCredentials:
+    """Helper (frozen dataclass) for storing temporary ECR credentials."""
+
+    username: str
+    password: str
+    proxy_endpoint: str
+    expires_at: datetime
+
+    def __post_init__(self):
+        mask_secret(self.password)
+        logger.debug("Credentials to Amazon ECR %r expires at %s.", self.proxy_endpoint, self.expires_at)
+
+    @property
+    def registry(self) -> str:
+        """Return registry in appropriate `docker login` format."""
+        # https://github.com/docker/docker-py/issues/2256#issuecomment-824940506
+        return self.proxy_endpoint.replace("https://", "")
+
+
+class EcrHook(AwsBaseHook):
+    """
+    Interact with Amazon Elastic Container Registry (ECR)
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    def __init__(self, **kwargs):
+        kwargs.pop("client_type", None)
+        super().__init__(client_type="ecr", **kwargs)
+
+    def get_temporary_credentials(self, registry_ids: list[str] | str | None = None) -> list[EcrCredentials]:
+        """Get temporary credentials for Amazon ECR.
+
+        Return list of :class:`~airflow.providers.amazon.aws.hooks.ecr.EcrCredentials`,
+        obtained credentials valid for 12 hours.
+
+        :param registry_ids: Either AWS Account ID or list of AWS Account IDs that are associated
+            with the registries for which obtain credentials. If you do not specify a registry,
+            the default registry is assumed.
+
+        .. seealso::
+            - `boto3 ECR client get_authorization_token method <https://boto3.amazonaws.com/v1/documentation/\
+               api/latest/reference/services/ecr.html#ECR.Client.get_authorization_token>`_.
+        """
+        registry_ids = registry_ids or None
```

Comment:
Is this just to guard empty string/list as an input? Because you are already defaulting this to None in the method signature.

Reply:
Yep, just a guard. This attribute will propagated from connection

Reply:
So value in some cases might be `""` just for avoid this situation.

Reply:
@o-nikolas @ferruzzi do we have any concern about this part? Or we could merge this changes?

Reply:
Yeah no concern I was just checking to be sure. You could re-arrange the if statements so that the falsey check happens first so that you wouldn't need this assignment, but the way you have it is also perfectly fine :+1: